### PR TITLE
Use SYSTEMROOT env variable for powershell detection

### LINKED ISSUE #29802: Hard-coded path in Base.download to PowerShell may fail
In v1.0.1, [base/download.jl](https://github.com/JuliaLang/julia/blob/0d713926f85dfa3e4e0962215b909b8e47e94f48/base/download.jl#L7) uses a hard-coded path to locate `powershell.exe`:
```julia
5 if Sys.iswindows()
6   function download(url::AbstractString, filename::AbstractString)
7     ps = "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
```

This fails if `C:\Windows` is not the Windows directory. One consequence is that Pkg may fail to download artifacts during an `add` operation.

It may be more robust to use [`WINDIR`](https://docs.microsoft.com/en-us/windows/deployment/usmt/usmt-recognized-environment-variables):
```julia
5 if Sys.iswindows()
6   function download(url::AbstractString, filename::AbstractString)
7     ps = "%WINDIR%\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
```

### PR DESCRIPTION
**Pull request by musm**
fix https://github.com/JuliaLang/julia/issues/29802

### CODE REVIEW

File: base/download.jl [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 if Sys.iswindows()
     function download(url::AbstractString, filename::AbstractString)
-        ps = "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
+        ps = joinpath(ENV["WINDIR"], "System32\\WindowsPowerShell\\v1.0\\powershell.exe")
```
Comment (vtjnash) [inline base/download.jl]: use `get(ENV, "WINDIR", "C:\\Windows")`

Comment (musm) [reply base/download.jl]: I guess it's safer but WINDIR is defined by the system

File: base/download.jl [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 if Sys.iswindows()
     function download(url::AbstractString, filename::AbstractString)
-        ps = "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe"
+        ps = joinpath(get(ENV, "SYSTEMROOT", "C:\\Windows"), "System32\\WindowsPowerShell\\v1.0\\powershell.exe")
```
Comment (musm) [inline base/download.jl]: @vtjnash  are you ok with removing the fallback to `C:\\Windows` , I think it's superfluous 

You would have much bigger problems  than  anything to do with Julia if someone fudged this env variable

Comment (nalimilan) [reply base/download.jl]: Conversely, it there any reason not to keep it?

Comment (musm) [reply base/download.jl]: Well if SYSTEMROOT didn't' work to find powershell, there's no way "C:\\Windows" would either, so I think it's superfluous (not a big deal, but just saying).

**Comment by musm** [comment]
merge?  (backport tag as well)

**Comment by musm** [comment]
bump

**Comment by StefanKarpinski** [comment]
@vtjnash: this has been updated—if it looks good to you now, merge?

**Comment by musm** [comment]
also @staticfloat (same change as in binarybuilder)

**Review by staticfloat — APPROVED** [review]

**Comment by musm** [comment]
thanks for the approve and merge.  Can we add a backport label here too?

**Comment by musm** [comment]
@KristofferC  this is a backport eligible  PR

**Pull request merged**